python/lsst/dbb/buffmngrs/endpoint/ingester/ingester.py

```diff
@@ -182,16 +182,23 @@ def run(self):
             #
             # (a) whose name matches at least one pattern on the include list,
             # (b) does not match any pattern on the exclude list,
-            # (c) is not an empty.
+            # (c) still exists (a sanity check),
+            # (d) is not empty.
             #
-            # Only the outcome of the last failed check is stored in
-            # the database.
+            # If a file fails to pass a check, any remaining checks will be
+            # skipped.
+            #
+            # We don't want to create events, other than about being ignored,
+            # for files which don't meet the search criteria. Thus matching
+            # a filename against the patterns on the include and the exclude
+            # lists should precede any other checks.
             for rec in records:
                 path = os.path.join(rec.relpath, rec.filename)
+                abspath = os.path.join(self.storage, path)
                 size = rec.size_bytes
```

Comment:
I would probably have moved this down near the code that uses it (or just explicitly use rec.size_bytes in the code).